From akpm@linux-foundation.org  Tue Sep 23 13:19:08 2008
From: David Almaroad <dalmaroad@gmail.com>
Date: Mon, 22 Sep 2008 15:00:11 -0700
Subject: usb: unusual devs patch for Nokia 5310 Music Xpress
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org, akpm@linux-foundation.org, dalmaroad@gmail.com, mdharm-usb@one-eyed-alien.net
Message-ID: <200809222200.m8MM0B3e032607@imap1.linux-foundation.org>


From: David Almaroad <dalmaroad@gmail.com>

The Nokia 5310 Music Xpress phone reports one too many sectors in
usb-storage mode.  This patch resolves that.

Signed-off-by: David Almaroad <dalmaroad@gmail.com>
Cc: Matthew Dharm <mdharm-usb@one-eyed-alien.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/unusual_devs.h |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/drivers/usb/storage/unusual_devs.h
+++ b/drivers/usb/storage/unusual_devs.h
@@ -246,6 +246,13 @@ UNUSUAL_DEV(  0x0421, 0x04fa, 0x0601, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_CAPACITY ),
 
+/* Patch for Nokia 5310 capacity */
+UNUSUAL_DEV(  0x0421, 0x006a, 0x0000, 0x0591,
+	"Nokia",
+	"5310",
+	US_SC_DEVICE, US_PR_DEVICE, NULL,
+	US_FL_FIX_CAPACITY ),
+
 /* Reported by Olaf Hering <olh@suse.de> from novell bug #105878 */
 UNUSUAL_DEV(  0x0424, 0x0fdc, 0x0210, 0x0210,
 		"SMSC",
